src/Sentry/SentryClient.cs

```diff
@@ -122,24 +122,39 @@ public void CaptureTransaction(Transaction transaction)
             if (string.IsNullOrWhiteSpace(transaction.Name) ||
                 string.IsNullOrWhiteSpace(transaction.Operation))
             {
-                _options.DiagnosticLogger?.LogWarning("Transaction discarded due to one or more required fields missing.");
+                _options.DiagnosticLogger?.LogWarning(
+                    "Transaction discarded due to one or more required fields missing."
+                );
+
                 return;
             }
 
-            // A transaction may have already been sampled somehow or the
-            // field may have been set directly. To be safe, we check that.
-            if (!transaction.IsSampled)
+            // If a transaction has been already sampled out (by adaptive sampling, at the time
+            // the transaction was created), then we just drop it here.
+            if (transaction.IsSampled == false)
             {
-                _options.DiagnosticLogger?.LogDebug("Transaction dropped due to sampling.");
+                _options.DiagnosticLogger?.LogDebug(
+                    "Transaction dropped due to adaptive sampling."
```

Comment:
Was it due to adaptive sampling? Maybe better:

```suggestion
                    "Transaction dropped due to being marked as unsampled."
```